conda/lock.py

```diff
@@ -61,11 +73,44 @@ def __enter__(self):
                 return self
 
         stdoutlog.error("Exceeded max retries, giving up")
-        raise LockError(lockstr % self.lock_path)
+        raise LockError(LOCKSTR % self.lock_path)
 
     def __exit__(self, exc_type, exc_value, traceback):
         try:
             os.rmdir(self.lock_path)
             os.rmdir(self.path)
         except OSError:
             pass
+
+
+class FileLock(object):
+    """
+    Context manager to handle locks.
+    """
+    def __init__(self, filepath, retries=10):
+        self.filepath = abspath(filepath)
+        self.retries = retries
+
+    def __enter__(self):
+        sleeptime = 1
+        self.lockpath = "{0}.pid{1}.{2}".format(self.filepath, os.getpid(), LOCKFN)
+        lockglobstr = "{0}.pid*.{2}".format(self.filepath, LOCKFN)
+        lastglobmatch = None
+        for _ in range(self.retries):
+            globresult = glob(lockglobstr)
+            if globresult:
+                stdoutlog.info(LOCKSTR % globresult[0])
+                stdoutlog.info("Sleeping for %s seconds\n" % sleeptime)
+
+                time.sleep(sleeptime/10)
+                sleeptime *= 2
+                lastglobmatch = globresult
+            else:
+                touch(lockpath)
+                return self
+
+        stdoutlog.error("Exceeded max retries, giving up")
+        raise LockError(LOCKSTR % lastglobmatch[0])
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        rm_rf(self.lockpath)
```

Comment:
`rm_rf` comes from `conda/install.py`.  That doesn't make any sense.  As I said above, all common "file utility" functions need their own module.  We'll rearrange sometime soon.